easybuild/framework/easyconfig/tools.py

```diff
@@ -248,12 +249,16 @@ def det_easyconfig_paths(orig_paths, from_pr=None, easyconfigs_pkg_paths=None):
     if easyconfigs_pkg_paths is None:
         easyconfigs_pkg_paths = []
 
+    pr_files = None
+    if from_pr is not None:
+        pr_files = fetch_easyconfigs_from_pr(from_pr)
+
     ec_files = orig_paths[:]
 
-    if not ec_files and from_pr:
-        pr_files = fetch_easyconfigs_from_pr(from_pr)
+    if not ec_files and pr_files:
         ec_files = [path for path in pr_files if path.endswith('.eb')]
-
+    elif ec_files and pr_files:
+        ec_files = [path for path in pr_files if os.path.basename(path) in orig_paths]
```

Comment:
Hmm, this is different than what I would do...

Here, you only retain files from the PR that are also listed on the command line. So, any files that are listed but not in the PR will be totally ignored, which is wrong imho. It's also limiting. What if the PR only touches files that are dependencies for X, and I want to make sure X doesn't break with the rebuilt dependencies as specified by the PR?